tests/pruners_tests/test_successive_halving.py

```diff
@@ -253,10 +256,15 @@ def test_successive_halving_pruner_min_early_stopping_rate_parameter() -> None:
 def test_successive_halving_pruner_bootstrap_parameter() -> None:
 
     with pytest.raises(ValueError):
-        optuna.pruners.SuccessiveHalvingPruner(bootstrap_count=-1)
+        # Because the default min_resource is "auto", the bootstrap_count must be zero.
+        # The type checker correctly detects this type error.
+        optuna.pruners.SuccessiveHalvingPruner(bootstrap_count=-1)  # type: ignore
```

Comment:
Maybe after we are able to detect these errors by type-checking, we could remove these tests according to the test policy.

Reply:
@ConnorBaker We concluded that we can remove these invalid tests in this PR if we can detect them by type-checking. How about that?